nslsii/temperature_controllers.py

```diff
@@ -35,29 +37,44 @@ def __init__(self, pv_prefix, **kwargs):
     # define the new set method with the new moving indicator
     def set(self, value):
         # check that a set is not in progress, and if not set the lock.
-        if self._set_lock:
-            raise Exception('attempting to set {}'.format(self.name) +
+        if not self._set_lock.acquire(blocking=False):
+            raise Exception('attempting to set {} '.format(self.name) +
```

Comment:
While we're polishing this, I'll mention a possible improvement that I withheld previously. When a library raises `Exception`, the only way for the call to catch that exception (in order to display a message in a GUI or re-raise a different error, or try again...) is to catch it with `raise Exception` which is overly blunt and might catch many things. Better to be more specific. This could be, for example, a `RuntimError` or -- better yet -- something quite narrow. This design philosophy can be taken too far, of course, but in this case I would suggest:

```
class SetInProgress(RuntimeError):
   ...
```

Reply:
Also changed